### ISSUE TITLE
Trainer progress bar not showing consistently

### ISSUE THREAD

**Issue by srv902** [chainer/chainer#3557]
Hi,

I am running a simple NN model for regression. So when I set the epochs (say 1000) in the trainer.Trainer function, it runs and shows the progress bar. Now I am loading the model with or without the same name and if I continue with the same number of epochs (1000), progress bar is not displayed, but if I change it to 1100 or any number more than 1000 it runs and shows progress bar. Also when I restart the Jupyter Notebook and execute trainer.run(), it shows progress bar for 1000 epochs.
   
    class NNTest(link.Chain):

           def __init__(self,nunits,nout):
                 super(NNTest,self).__init__()
                 with self.init_scope():
                       self.l1 = L.Linear(None,nunits)
                       self.l2 = L.Linear(None,nunits)
                       self.l3 = L.Linear(None,1)
                     
           def __call__(self,x,ytrue):
                 h1 = F.relu(self.l1(x))
                 h2 = F.relu(self.l2(h1))
                 h3 = F.reshape(self.l3(h2),(28,))
                 loss = self.wapeloss(h3,ytrue)
                 report({'loss':loss},self)
                 return loss
    
           def wapeloss(self,ypred,ytrue):
                 return F.sum(abs(ypred-ytrue))/F.sum(ytrue)
    
           def predict(self,x1,y1):
                 t1 = self.l3(self.l2(self.l1(x1)))
                 t2 = np.reshape(t1,(x1.shape[0],))
                 print("WAPE value: ",calc_wape(y1.values,t2))
                 return t2

and the optimizer and training code

           model = NNTest(30,60) 
           optim1 = optimizers.Adam()
           optim1.setup(model)
           updater1 = training.StandardUpdater(train_iter, optim1)
           trainer1 = training.Trainer(updater1, (1000, 'epoch'), out='result')
           trainer1.extend(extensions.ProgressBar())

Chainer version is 2.1.0
Windows 10, Anaconda version 5.0.0 (64bit)

**Comment by delta2323** [comment]
Hi, thank you for reporting the issue and sorry for my late response. 

Did you save the trainer itself or just a model `NNTest`? Trainer remembers its epoch number when serialized and restores it when deserialized. So, if you serialize the trainer that runs 1000 epoch, load it, and make it run with the same epoch number. No training will occur (and therefore the progressbar will not be fired).

**Comment by kmaehashi** [comment]
If you still have issues, please provide the full notebook code (or how you deserialize model). Thank you.

**Comment by mitmul** [comment]
Could you check the above comments? @srv902

**Comment by kmaehashi** [comment]
Feel free to reopen the issue if you still have a problem.
